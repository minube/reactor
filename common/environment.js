import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
