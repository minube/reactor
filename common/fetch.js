const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
