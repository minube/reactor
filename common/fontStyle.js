import screenSize from './screenSize';
import THEME from './theme';

const { FONT, UNIT } = THEME;

export default (
  width,
  {
    tiny, small, large, title,
  },
) => {
  const {
    TINY, PHONE, TABLET, SMALL, REGULAR, LARGE,
  } = screenSize(width);

  let fontSize = REGULAR || LARGE ? FONT.SIZE.REGULAR : FONT.SIZE.SMALL;

  if (tiny) fontSize = REGULAR || LARGE ? FONT.SIZE.TINY : UNIT * 1.1;
  else if (small) fontSize = REGULAR || LARGE ? FONT.SIZE.SMALL : FONT.SIZE.TINY;
  else if (large) {
    fontSize = FONT.SIZE.LARGE;
    if (TINY) fontSize = FONT.SIZE.SMALL;
    if (PHONE || SMALL) fontSize = FONT.SIZE.REGULAR;
    if (TABLET || REGULAR) fontSize = UNIT * 1.8;
  } else if (title) {
    fontSize = UNIT * 4.8;
    if (TINY || PHONE || TABLET) fontSize = UNIT * 2.4;
    if (SMALL || REGULAR) fontSize = UNIT * 3;
  }

  return {
    fontSize,
    lineHeight: fontSize * (TINY || SMALL ? 1.2 : 1.3),
  };
};
