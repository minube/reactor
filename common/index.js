import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
