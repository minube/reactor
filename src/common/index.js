import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
