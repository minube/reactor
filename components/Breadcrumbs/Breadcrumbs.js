import { array, arrayOf, func, number, oneOfType, shape, string } from 'prop-types';
import React from 'react';
import { StyleSheet, View } from 'react-native';

import { THEME } from '../../common';
import Link from '../Link';
import styles from './Breadcrumbs.style';

const { COLOR } = THEME;

const Breadcrumbs = ({
  color, dataSource, onPress, style,
}) => (
  <View style={styles.container}>
    { dataSource.map(({ caption, href }, index) => (
      <Link
        href={href}
        key={caption}
        onPress={() => {
          onPress(index);
          return false;
        }}
        style={StyleSheet.flatten([styles.link, { color }, style])}
        tiny
      >
        {`${caption}${index < dataSource.length - 1 ? ' Â·' : ''}`}
      </Link>
    ))}
  </View>
);

Breadcrumbs.propTypes = {
  color: string,
  dataSource: arrayOf(shape({})),
  onPress: func,
  style: oneOfType([array, number]),
};

Breadcrumbs.defaultProps = {
  color: COLOR.TEXT,
  dataSource: [],
  onPress() {},
  style: [],
};

export default Breadcrumbs;
