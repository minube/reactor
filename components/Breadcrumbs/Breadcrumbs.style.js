import { StyleSheet } from 'react-native';

import { STYLE, THEME } from '../../common';

const { UNIT } = THEME;

export default StyleSheet.create({
  link: {
    marginRight: UNIT / 2.25,
  },

  container: STYLE.ROW,
});
