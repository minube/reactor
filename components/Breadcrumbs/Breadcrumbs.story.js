import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { color, object } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Breadcrumbs from './Breadcrumbs';

const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: THEME.OFFSET };

const dataSource = [
  { caption: 'Spain', href: '/' },
  { caption: 'Basque Country', href: '/' },
  { caption: 'Bilbao', href: '/' },
];

storiesOf('✅ Breadcrumbs', module)
  .add('default', () => (
    <Breadcrumbs dataSource={dataSource} />
  ))
  .add('color', () => (
    <Breadcrumbs dataSource={dataSource} color={THEME.COLOR.PRIMARY} />
  ))
  .add('⚡ onPress', () => (
    <Breadcrumbs dataSource={dataSource} onPress={action('Breadcrumbs.onPress()')} />
  ))
  .add('style', () => (
    <Breadcrumbs dataSource={dataSource} style={STYLE} />
  ))
  .add('🏀 Playground', () => (
    <Breadcrumbs
      dataSource={object('dataSource', dataSource, 'heading-dataSource')}
      color={color('color', THEME.COLOR.TEXT)}
      style={object('style', STYLE)}
    />
  ));
