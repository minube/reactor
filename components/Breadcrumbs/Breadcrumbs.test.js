import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Breadcrumbs from './Breadcrumbs';

const dataSource = [
  { caption: 'Spain', href: '/' },
  { caption: 'Basque Country', href: '/' },
  { caption: 'Bilbao', href: '/' },
];

describe('<Breadcrumbs>', () => {
  it('renders', () => {
    const tree = renderer.create(<Breadcrumbs />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {dataSource}', () => {
    const tree = renderer.create(<Breadcrumbs dataSource={dataSource} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Breadcrumbs dataSource={dataSource} color={THEME.COLOR.PRIMARY} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onPress}', () => {
    // @TODO: Test <Breadcrumbs /> {onPress}
  });

  it('when {style}', () => {
    const style = {
      padding: THEME.OFFSET,
      backgroundColor: 'rgba(0,0,0,0.25)',
    };
    const tree = renderer.create(<Breadcrumbs dataSource={dataSource} style={style} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
