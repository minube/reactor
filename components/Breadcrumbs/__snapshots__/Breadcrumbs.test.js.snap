// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs> renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
/>
`;

exports[`<Breadcrumbs> when {color} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "color": "#10c0bc",
        "fontFamily": undefined,
        "fontSize": 14,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
      }
    }
  >
    Spain ·
  </Text>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "color": "#10c0bc",
        "fontFamily": undefined,
        "fontSize": 14,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
      }
    }
  >
    Basque Country ·
  </Text>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "color": "#10c0bc",
        "fontFamily": undefined,
        "fontSize": 14,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
      }
    }
  >
    Bilbao
  </Text>
</View>
`;

exports[`<Breadcrumbs> when {dataSource} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontSize": 14,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
      }
    }
  >
    Spain ·
  </Text>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontSize": 14,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
      }
    }
  >
    Basque Country ·
  </Text>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontSize": 14,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
      }
    }
  >
    Bilbao
  </Text>
</View>
`;

exports[`<Breadcrumbs> when {style} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.25)",
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontSize": 16,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
        "padding": 16,
      }
    }
  >
    Spain ·
  </Text>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.25)",
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontSize": 16,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
        "padding": 16,
      }
    }
  >
    Basque Country ·
  </Text>
  <Text
    accessibilityRole="link"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    href="/"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.25)",
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontSize": 16,
        "lineHeight": 14.84,
        "marginRight": 4.444444444444445,
        "padding": 16,
      }
    }
  >
    Bilbao
  </Text>
</View>
`;
